File: firebird/php/create/firebird_create.php
#! /usr/bin/php
<?php
// ------------------------------------------------------------------
//	firebird_create.php
//
//					May/09/2013
//
// ------------------------------------------------------------------
$path="/var/www/data_base/common/php_common";
set_include_path (get_include_path() . PATH_SEPARATOR . $path);

include "firebird_display.php";
include "firebird_row_update.php";
// ------------------------------------------------------------------
print	"*** 開始 ***\n";


define("DBNAME","localhost:/var/tmp/firebird/cities.fdb");
define("DBUSER","sysdba");
define("DBPASS","tiger");

$dbh = ibase_connect(DBNAME,DBUSER,DBPASS,"UTF-8");
if ($dbh == FALSE)
{
	echo 'データベースと接続できません。\n';
}
else
{
	print 'データベースとの接続に成功しました。' . "\n";

	firebird_drop_table_proc ($dbh);
	firebird_create_table_proc ($dbh);


	firebird_insert_proc ($dbh,'t3821','松山',75428,'2002-3-27');
	firebird_insert_proc ($dbh,'t3822','今治',54715,'2002-7-9');
	firebird_insert_proc ($dbh,'t3823','宇和島',23458,'2002-8-17');
	firebird_insert_proc ($dbh,'t3824','八幡浜',74682,'2002-6-18');
	firebird_insert_proc ($dbh,'t3825','新居浜',62541,'2002-4-24');
	firebird_insert_proc ($dbh,'t3826','西条',46957,'2002-8-21');
	firebird_insert_proc ($dbh,'t3827','大洲',37516,'2002-5-8');
	firebird_insert_proc ($dbh,'t3828','伊予',82592,'2002-10-1');
	firebird_insert_proc ($dbh,'t3829','西予',61745,'2002-12-12');

	firebird_display_proc ($dbh);
   	ibase_close($dbh);
}

print	"*** 終了 ***\n";
// ------------------------------------------------------------------
?>

File: firebird/perl/create/firebird_create.pl
#! /usr/bin/perl
#
#	firebird_create.pl
#
#					May/30/2012
#
#
# ----------------------------------------------------------------
use	strict;
use	warnings;
use	utf8;
use	Encode;
use	DBI;
#
use lib '/var/www/data_base/common/perl_common';
use text_manipulate;
use firebird_manipulate;
#
# ----------------------------------------------------------------
print	(encode ('utf-8',"*** 開始 ***\n"));
#
my $host     = 'localhost';
my $dbpath   = '/var/tmp/firebird/cities.fdb';
my $user     = 'SYSDBA';
my $password = 'tiger';
#
#
my $data_source = 'dbi:Firebird:' . "host=$host;dbname=$dbpath";

print	("*** pppp ***\n");
my $dbh = DBI->connect($data_source, $user, $password);
unless($dbh) {
	print STDERR "Error : connect.\n";
	exit -1;
}
#
my %dict_aa = data_prepare_proc ();
#
text_manipulate::dict_display_proc (%dict_aa);
#
firebird_manipulate::firebird_drop_proc ($dbh);
firebird_manipulate::firebird_create_proc ($dbh);
firebird_manipulate::firebird_table_insert_proc ($dbh,%dict_aa);
#
$dbh->disconnect;
#
print	(encode ('utf-8',"*** 終了 ***\n"));
#
# ----------------------------------------------------------------
sub data_prepare_proc
{
	my %dict_aa;
	%dict_aa = text_manipulate::dict_append_proc
		('t3821','松山',75842,'2005-9-30',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t3822','今治',14278,'2005-3-15',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t3823','宇和島',58641,'2005-6-17',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t3824','八幡浜',46132,'2005-8-22',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t3825','新居浜',27859,'2005-2-28',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t3826','西条',92581,'2005-5-9',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t3827','大洲',35497,'2005-4-10',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t3828','伊予',85736,'2005-10-8',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t3829','西予',64827,'2005-5-21',%dict_aa);

	return	%dict_aa;
}
# ----------------------------------------------------------------
